projects/common/src/time/user-specified-time-range.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core';
```

Comment:
This is the service responsible for maintaining the page user specified time ranges in preferences storage. The time ranges are all stored in one object, under one preference key - the page paths are used as the keys of the object.

Because they're stored under one object, the set method needs access to the existing stored TRs object (`pageTimeMap`) before setting the TR for the path it was called with - for this the service maintains a subject, `pageTimeRangesSubject$`, so the set method can synchronously access the object.